presto-main/src/main/java/com/facebook/presto/connector/system/SystemSplitManager.java

```diff
@@ -76,17 +78,26 @@ public ConnectorSplitSource getPartitionSplits(ConnectorTableHandle table, List<
         SystemTable systemTable = tables.get(systemPartition.getTableHandle().getSchemaTableName());
         checkArgument(systemTable != null, "Table %s does not exist", systemPartition.getTableHandle().getTableName());
 
-        if (systemTable.isDistributed()) {
-            ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
-            for (Node node : nodeManager.getActiveNodes()) {
-                splits.add(new SystemSplit(systemPartition.getTableHandle(), node.getHostAndPort()));
-            }
-            return new FixedSplitSource(SystemConnector.NAME, splits.build());
+        TableDistributionEnum tableDistributionMode = systemTable.getDistributionMode();
+        if (tableDistributionMode == TableDistributionEnum.SINGLE_COORDINATOR) {
+            HostAddress address = nodeManager.getCurrentNode().getHostAndPort();
```

Comment:
Is this intended to only allow querying the system tables from the coordinator? Wouldn't you need to check if this node is a coordinator? it's still possible to connect to a worker and issue queries.